src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -176,14 +180,16 @@ private SerializationResult serializeFromJavaObjects(
       rows.addAll(data.getVectors().rows);
 
       rowCount += data.getRowCount();
+      chunkUncompressedSize += data.getBufferSize();
```

Comment:
Same here, is it possible we get the actual uncompressed buffer size instead of using the estimated buffer size?

Reply:
Yes, `bufferSize()` is an estimation. For Parquet we have no access to the 'real' uncompressed data size because data is compressed internally. Our estimation can be slightly larger than the real value.